File: plop-templates/adapter/acquiring/src/handler/AuthHandler.java
package com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.handler;

import com.rbkmoney.adapter.common.enums.Step;
import com.rbkmoney.adapter.common.processor.Processor;
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.RemoteClient;
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.AuthRequest;
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.AuthResponse;
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.model.CustomEntryStateModel;
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.model.CustomExitStateModel;
import org.springframework.core.convert.converter.Converter;
import org.springframework.stereotype.Component;

@Component
public class AuthHandler extends StepHandler<AuthRequest, AuthResponse> {

    public AuthHandler(RemoteClient client,
                       Converter<CustomEntryStateModel, AuthRequest> converter,
                       Processor<CustomExitStateModel, AuthResponse, CustomEntryStateModel> processor) {
        super(client::auth, converter, processor);
    }

    @Override
    public boolean isHandle(CustomEntryStateModel entryStateModel) {
        return entryStateModel.getAdapterContext().getStep() == Step.AUTH;
    }

}

File: plop-templates/adapter/acquiring/src/handler/AuthHandler.kt
package com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.handler

import com.rbkmoney.adapter.common.enums.Step
import com.rbkmoney.adapter.common.processor.Processor
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.RemoteClient
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.AuthRequest
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.client.model.AuthResponse
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.model.CustomEntryStateModel
import com.rbkmoney.{{adapterAcquiringPackageCase bank_name}}.model.CustomExitStateModel
import org.springframework.core.convert.converter.Converter
import org.springframework.stereotype.Component

@Component
class AuthHandler(
    client: RemoteClient,
    converter: Converter<CustomEntryStateModel, AuthRequest>,
    processor: Processor<CustomExitStateModel, AuthResponse, CustomEntryStateModel>
) : StepHandler<AuthRequest, AuthResponse>(client::auth, converter, processor) {

    override fun isHandle(entryStateModel: CustomEntryStateModel): Boolean {
        return entryStateModel.adapterContext.step == Step.AUTH
    }
}
